lms/djangoapps/courseware/management/commands/fix_student_module_newlines.py

```diff
@@ -0,0 +1,330 @@
+        """Fix newline courses in CSM!"""
+        if len(args) != 2:
+            raise CommandError('Must specify start and end dates: e.g. "2016-08-23 16:43:00" "2016-08-24 22:00:00"')
+
+        start, end = args
+        dry_run = options['dry_run']
+
+        log.info(
+            "Starting fix_student_module_newlines in %s mode!",
+            "dry_run" if dry_run else "real"
+        )
+
+        rows_to_fix = use_read_replica_if_available(
+            # pylint: disable=no-member
+            StudentModule.objects.raw(
+                "select * from courseware_studentmodule where modified between %s and %s and course_id like %s",
+                (start, end, '%\n')
+            )
+        )
+
+        results = [self.fix_row(row, dry_run=dry_run) for row in rows_to_fix]
+        log.info(
+            "Finished fix_student_module_newlines in %s mode!",
+            "dry_run" if dry_run else "real"
+        )
+        log.info("Stats: %s rows detected", len(results))
+        if results:
+            # Add up all the columns
+            aggregated_result = FixResult(*[sum(col) for col in zip(*results)])
+            log.info("Results: %s", aggregated_result)
+
+    def fix_row(self, read_only_newline_course_row, dry_run=False):
+        """
+        Fix a StudentModule with a trailing newline in the course_id.
+
+        Returns a count of (row modified, )
+
+        At the end of this method call, the record should no longer have a
+        trailing newline for the course_id. There are three possible outcomes:
+
+        1. There was never a conflicting entry:
+            -> We just update this row.
+        2. Conflict and the other row (with correct course_id) wins:
+            -> We delete this row.
+        2. Conflict and this row wins:
+            -> We copy the data to the conflicting row (the one that has a
+               correct course_id), and delete this row.
+
+        Even though all the StudentModule entries coming in here have trailing
+        newlines in the course_id, the deserialization logic will obscure that
+        (it gets parsed out when read from the database). We will also
+        automatically strip the newlines when writing back to the database, so
+        we have to be very careful about violating unique constraints by doing
+        unintended updates. That's why we only do an update to
+        newline_course_row if no correct_course_row exists.
+        """
+        # pylint: disable=too-many-statements
+        # We got the StudentModule from the read replica, so we have to fetch it
+        # again from our writeable database before making changes
+        try:
+            newline_course_row = StudentModule.objects.get(id=read_only_newline_course_row.id)
+        except StudentModule.DoesNotExist:
+            # We're not going to be able to make any corrective writes, so just fail fast
+            log.exception("Could not find editable CSM row %s", read_only_newline_course_row.id)
```

Comment:
If it's easiest, instead of updating all the current log messages, perhaps just adding a single log message with information about the course-id at the top of `fix_row`?